src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_primitive_fusing.cpp

```diff
@@ -687,9 +687,58 @@ void prepare_primitive_fusing::fuse_simple_primitives(program &p) {
             return true;
         };
 
+        auto try_fuse_through = [&](program_node& node) -> std::vector<program_node*> {
```

Comment:
Is there any reason to customize fusion logic instead of simple movements of primitives? 
`Prim -> Reshape -> FuseCandidate`  => `Prim -> FuseCandidate -> Reshape` => common fusion logic, etc

Reply:
Of course, this may also be a solution as well. But in some cases we don't know whether primitive will be fused or not (because of it's parameters/input or output data types/formats), so in such cases using independent pass will require copying fusions rules for primitive from prepare_primitive_fusings pass to make a decision about moving primitive. It will also require extra iterating through all primitives in topology. Additionally, such fusions require data type propogation (e.g. in case of quantize fusion) that is more convenient to do directly during the fusion process. And I thought that this feature is logically closer to existing fusions logic, so that is why I decided to extend current approach...

Reply:
> But in some cases we don't know whether primitive will be fused or not (because of it's parameters/input or output data types/formats

Do we really need to know that? I think the issue could happen for concat, where you have to create multiple nodes instead of single one, but for now only reshape and reorder are supported, so I don't expect significant perf difference in most of the cases due to swap itself. 
So I suggest doing experiment with such approach w/o complex layout/precision/parameters check logic.